src/workerd/util/duration-alert.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include <kj/string.h>
+#include <kj/time.h>
+#include <kj/debug.h>
+
+namespace workerd::util {
+
+// This is a utility class for instantiating a timer, which will log if it's destructed after a specified time
+// This works by relying on RAII (Scope-Bound Resource Management) to check how much time has elapsed
+// when the object is destructed. This ensures that the time is checked when the timer object goes out of scope
+// thereby timing everything after the initialization of the timer, until the end of the scope where it was
+// instantiated.
+class DurationAlert {
```

Comment:
Optional, non-blocking, nit: consider changing the name of this to DurationLogger or LongDurationLogger since that better describes what the class does.  (Alerts mean different things.)